watercore-base/src/main/java/me/lucyy/watercore/api/impl/data/MemoryDataStore.java

```diff
@@ -1,40 +1,21 @@
-/*
- * Copyright © 2021 Lucy Poulton.
- * This file is part of watercore.
- *
- * watercore is free software: you can redistribute it and/or modify
- * it under the terms of the GNU General Public License as published by
- * the Free Software Foundation, either version 3 of the License, or
- * (at your option) any later version.
- *
- * watercore is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with watercore.  If not, see <https://www.gnu.org/licenses/>.
- */
-
 package me.lucyy.watercore.api.impl.data;
 
 import me.lucyy.watercore.api.data.DataStore;
-import me.lucyy.watercore.api.data.DataKey;
 import org.jetbrains.annotations.Nullable;
 import java.io.Serializable;
 import java.util.HashMap;
 
 public class MemoryDataStore implements DataStore {
-	private final HashMap<DataKey<? extends Serializable>, Serializable> storeMap = new HashMap<>();
+	private final HashMap<Key<? extends Serializable>, Serializable> storeMap = new HashMap<>();
```

Comment:
Use Map in place of HashMap when possible. So this turns into 
`Map<Key<? extends Serializable>, Serializable> storeMap = new HashMap<>()`
instead of
`HashMap<Key<? extends Serializable>, Serializable> storeMap = new HashMap<>()`.
This allows it so there is less restriction over the general API itself.